plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -474,6 +488,35 @@ public void downloadPlugins(List<Plugin> plugins) {
                 e.printStackTrace();
             }
         }
+
+        // Filter out failed plugins
+        final List<Plugin> failedPlugins = getFailedPlugins();
+        if (!skipFailedPlugins && failedPlugins.size() > 0) {
+            throw new DownloadPluginException("Some plugin downloads failed: " +
+                    failedPlugins.stream().map(Plugin::getName).collect(Collectors.joining(",")));
+        }
+        Set<String> failedPluginNames = new HashSet<>(failedPlugins.size());
+        failedPlugins.forEach(plugin -> failedPluginNames.add(plugin.getName()));
+
+        // Copy files over to the destination directory
+        for (Plugin plugin : plugins) {
+            String archiveName = plugin.getArchiveFileName();
+            File downloadedPlugin = new File(downloadsTmpDir, archiveName);
+            try {
+                if (failedPluginNames.contains(plugin.getName())) {
+                    System.out.println("Will skip the failed plugin download: " + plugin.getName());
+                    Files.deleteIfExists(downloadedPlugin.toPath());
```

Comment:
should we have an option to keep failed downloads for inspection? even an env variable?

Reply:
If `skipFailedPlugins ` is not set, it will be interrupted, and the files will be available for inspection. Not sure we want to keep them if we skip downloads, but I am happy to leave the, in tmp